driver.go

```diff
@@ -196,22 +196,81 @@ func (d *Driver) SetConfig(cfg *base.Config) error {
 		timeout = t
 	}
 
-	d.podman = d.newPodmanClient(timeout)
-	d.slowPodman = d.newPodmanClient(0)
+	if len(d.config.Socket) > 0 && d.config.SocketPath != "" {
+		return fmt.Errorf("error: can't define socket blocks and socket_path, they're mutually exclusive.")
+	} else if len(d.config.Socket) > 0 {
+		d.podmanClients = d.makePodmanClients(d.config.Socket, timeout)
+	} else if d.config.SocketPath != "" {
+		d.podmanClients = make(map[string]*api.API)
+		d.podmanClients["default"] = d.newPodmanClient(timeout, d.config.SocketPath, "", true)
+		d.defaultPodman = d.podmanClients["default"]
+	} else {
+		d.podmanClients = make(map[string]*api.API)
+		uid := os.Getuid()
+		if uid == 0 {
+			d.podmanClients["default"] = d.newPodmanClient(timeout, "unix:///run/podman/podman.sock", "root", true)
+		} else {
+			d.podmanClients["default"] = d.newPodmanClient(timeout, fmt.Sprintf("unix:///run/user/%d/podman/podman.sock", uid), "", true)
+		}
+		d.defaultPodman = d.podmanClients["default"]
+	}
 	d.compute = cfg.AgentConfig.Compute()
 
 	return nil
 }
 
+func (d *Driver) makePodmanClients(sockets []PluginSocketConfig, timeout time.Duration) map[string]*api.API {
```

Comment:
In various parts of Nomad core, we've handled this "default config block" problem by requiring the _name_ of the default config be "default" and interpolate the `name = ""` as `name = "default"`. That reduces the amount of logic you need here and lets users omit the name field if they have only a single default config.

Reply:
I've changed the implementation as described. It would've been nicer if I was able to push some of the logic to the HCL config parser (for example demanding no duplicate names, limiting the characters that are allowed, ...)

Reply:
> It would've been nicer if I was able to push some of the logic to the HCL config parser

Yeah, we don't really provide a good "driver SDK" for that kind of thing.